### PR TITLE
avoid adding lib subdirs to $*LIBRARY_PATH if no libraries are there in PythonPackage easyblock

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
@wpoely86: please review?

File: easybuild/easyblocks/generic/pythonpackage.py [hunk 1]
```diff
@@ -294,6 +294,25 @@ def sanity_check_step(self, *args, **kwargs):
...
+        for envvar in ['LD_LIBRARY_PATH', 'LIBRARY_PATH']:
+            newlist = []
+            for subdir in guesses[envvar]:
+                # only subdirectories that contain one or more files/libraries should be retained
+                fullpath = os.path.join(self.installdir, subdir)
+                if os.path.exists(fullpath) and any([os.path.isfile(x) for x in os.listdir(fullpath)]):
+                    newlist.append(subdir)
+            guesses[envvar] = newlist
+
+        return guesses
```
Comment (wpoely86) [inline easybuild/easyblocks/generic/pythonpackage.py]: maybe add a debug print about which directories are retained?

**Comment by wpoely86** [comment]
lgtm

File: easybuild/easyblocks/generic/pythonpackage.py (hunk 1 above)
Comment (boegel) [inline easybuild/easyblocks/generic/pythonpackage.py]: ok, will do, thanks

Comment (boegel) [inline easybuild/easyblocks/generic/pythonpackage.py]: debug log now includes something like:

```
== 2015-11-27 10:26:36,023 runpy.PythonPackage DEBUG Only retaining [] subdirs from ['lib', 'lib64', 'lib32'] for $LIBRARY_PATH (others don't provide any libraries)
```

**Comment by boegel** [comment]
Good to go after some further testing, thanks for the review @wpoely86!

**Pull request merged**